src/LogikUI/File/FileManager.cs

```diff
@@ -0,0 +1,337 @@
+                }
+                else
+                {
+                    MessageDialog md = new MessageDialog(null, DialogFlags.DestroyWithParent, MessageType.Error, ButtonsType.Ok, "Logik is unable to access choosen file.");
+                    md.Run();
+                    md.Dispose();
+
+                    throw new ArgumentException("Unable to access file.", e);
+                }
+            }
+
+            if (reader != null) reader.Close();
+        }
+
+        /// <summary>
+        /// Writes the project to an earlier specified file.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">No target file has been specified.</exception>
+        public static void Save()
+        {
+            if (IsNew) throw new InvalidOperationException("The file is new.");
+
+            //FIXME: delete file before save
+            
+            Save(filename);
+
+        }
+
+        // <summary>
+        // Writes the project to a specified file.
+        // </summary>
+        // <param name="filename">The path of the new project file.</param>
+        public static void Save(string filename)
+        {
+           
+            IsNew = false;
+            FileManager.filename = filename;
+
+            try
+            {
+                XmlWriterSettings settings = new XmlWriterSettings();
+                settings.Indent = true;
+                settings.NewLineOnAttributes = true;
+
+                XmlWriter xmlWriter = XmlWriter.Create(filename, settings);
+                xmlWriter.WriteStartDocument();
+                xmlWriter.WriteStartElement("circuit");
+
+                #region Parse Wires
+
+                xmlWriter.WriteStartElement("wires");
+                foreach (var wire in Wires)
+                {
+
+                    xmlWriter.WriteStartElement("wire");
+
+                    //from
+                    xmlWriter.WriteStartElement("from");
+                    xmlWriter.WriteAttributeString("x", wire.Pos.X.ToString());
+                    xmlWriter.WriteAttributeString("y", wire.Pos.Y.ToString());
+                    xmlWriter.WriteEndElement();
+
+                    //to
+                    xmlWriter.WriteStartElement("to");
+                    xmlWriter.WriteAttributeString("x", wire.EndPos.X.ToString());
+                    xmlWriter.WriteAttributeString("y", wire.EndPos.Y.ToString());
+                    xmlWriter.WriteEndElement();
+
+                    xmlWriter.WriteEndElement();
+
+                }
+               
+                xmlWriter.WriteEndElement();
+
+                #endregion
+
+                #region Parse Components
+
+                xmlWriter.WriteStartElement("components");
+
+                foreach (var component in Components)
+                {
+                    xmlWriter.WriteStartElement("component");
+                    
+                    xmlWriter.WriteEndElement();
+                }
```

Comment:
I'm guessing this is where the writing of components is going to occur?
We can have a follow-up pr fix this.